### ISSUE TITLE
Missing property suggestion in map-style constructor invocation code assist when such property is defined just by its setter

### ISSUE THREAD

**Issue by mauromol** [groovy/groovy-eclipse#330]
This is a regression I discovered while re-testing the old GRECLIPSE-1772.

Consider the following:

```groovy
package c
class MyBean {
	String foo
	void setBar(String bar) {
		foo = bar.toUpperCase()
	}
}
```

And: 

```groovy
package c
class ConstructorTest {
	static main(args) {
		MyBean b = new MyBean(|)
	}
}
```

Invoke code assist at "|": the property `MyBean.bar` is not suggested as a possible map-style constructor argument, only `MyBean.foo` is.

Also, I personally think that the code assist should not automatically add the trailing ",", I would find more comfortable to decide myself if I really need a comma (to enter another key-value pair).

**Comment by eric-milles** [comment]
Could you give this a re-try with the latest build?

**Comment by mauromol** [comment]
This use case seems to be fixed in 2.9.2.xx-201709052200-e46, thank you! 👍
